android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt

```diff
@@ -36,4 +36,46 @@ class PatientExtensionTest {
 
     Assert.assertEquals("12 B, Gulshan, Karimabad Sindh", patient.extractAddress())
   }
+
+  @Test
+  fun testGetAgeString() {
+    val expectedAge = "1y"
+    Assert.assertEquals(expectedAge, getAgeStringFromDays(365))
+
+    val expectedAge2 = "1y 1m"
+    // passing days value for 1y 1m 4d
+    Assert.assertEquals(expectedAge2, getAgeStringFromDays(399))
+
+    val expectedAge3 = "1y"
+    // passing days value for 1y 1w
+    Assert.assertEquals(expectedAge3, getAgeStringFromDays(372))
+
+    val expectedAge4 = "1m"
+    Assert.assertEquals(expectedAge4, getAgeStringFromDays(35))
+
+    val expectedAge5 = "1m 2w"
+    Assert.assertEquals(expectedAge5, getAgeStringFromDays(49))
+
+    val expectedAge6 = "1w"
+    Assert.assertEquals(expectedAge6, getAgeStringFromDays(7))
+
+    val expectedAge7 = "1w 2d"
+    Assert.assertEquals(expectedAge7, getAgeStringFromDays(9))
+
+    val expectedAge8 = "3d"
+    Assert.assertEquals(expectedAge8, getAgeStringFromDays(3))
+
+    val expectedAge9 = "1y 2m"
+    Assert.assertEquals(expectedAge9, getAgeStringFromDays(450))
+
+    val expectedAge10 = "40y 3m"
+    Assert.assertNotEquals(expectedAge10, getAgeStringFromDays(14700))
+
+    val expectedAge11 = "40y"
+    Assert.assertEquals(expectedAge11, getAgeStringFromDays(14700))
+
+    val expectedAge12 = ""
+    // if no age is map to patient model, or passing age as 0 days
+    Assert.assertEquals(expectedAge12, getAgeStringFromDays(0))
```

Comment:
@f-odhiambo for current modules it might be fine, however, as we progress with features and have newborn registrations as part of app then we may want to show age of an infant born on current date... So in that case age should be shown as 0d if birthdate is present and should be empty only if DoB is missing

Reply:
updated for 0d, when extractAge evaluates 0 days differences between currentDate and DOB

Reply:
@maimoonak